matRad_calcGeoDists.m

```diff
@@ -95,14 +98,23 @@
 ix = radDepthIx(subsetMask);
 
 % return radial distances squared
-if nargout > 1
-    rad_distancesSq = rad_distancesSq(subsetMask);
-end
+rad_distancesSq = rad_distancesSq(subsetMask);
 
 % return x & z distance
-if nargout > 2
-   isoLatDistsX = latDistsX(subsetMask)./rot_coords_temp(subsetMask,2)*SAD;
-   isoLatDistsZ = latDistsZ(subsetMask)./rot_coords_temp(subsetMask,2)*SAD; 
-end
-
+% if nargout > 2
+%    isoLatDistsX = latDistsX(subsetMask)./rot_coords_temp(subsetMask,2)*SAD;
+%    isoLatDistsZ = latDistsZ(subsetMask)./rot_coords_temp(subsetMask,2)*SAD; 
+% end
 
+% compute output only when requested (save computation time)
+if nargout > 4
+    % latDists
+    latDistsX = latDistsX(subsetMask);
+    latDistsZ = latDistsZ(subsetMask);
+    isoLatDistsX    = NaN;
+    isoLatDistsZ    = NaN;
+elseif nargout > 2
```

Comment:
The reason for this is to compute certain output only when specifically requested, but I see that it can make sense to request everything at the same time. I will change it accordingly.